src/core/lib/event_engine/tcp_socket_utils.cc

```diff
@@ -111,36 +111,13 @@ absl::StatusOr<std::string> ResolvedAddrToUnixPathIfPossible(
 
 absl::StatusOr<std::string> ResolvedAddrToUriUnixIfPossible(
     const EventEngine::ResolvedAddress* resolved_addr) {
-  const sockaddr* addr = resolved_addr->address();
-  if (addr->sa_family != AF_UNIX) {
-    return absl::InvalidArgumentError(
-        absl::StrCat("Socket family is not AF_UNIX: ", addr->sa_family));
-  }
-  const sockaddr_un* unix_addr = reinterpret_cast<const sockaddr_un*>(addr);
-#ifdef GPR_APPLE
-  int len = resolved_addr->size() - sizeof(unix_addr->sun_family) -
-            sizeof(unix_addr->sun_len) - 1;
-#else
-  int len = resolved_addr->size() - sizeof(unix_addr->sun_family) - 1;
-#endif
-  bool abstract = (len < 0 || unix_addr->sun_path[0] == '\0');
-  std::string scheme;
-  std::string path;
-  if (abstract) {
-    scheme = "unix-abstract";
-    if (len >= 0) {
-      path = std::string(unix_addr->sun_path + 1, len);
-    }
-  } else {
-    scheme = "unix";
-    size_t maxlen = sizeof(unix_addr->sun_path);
-    if (strnlen(unix_addr->sun_path, maxlen) == maxlen) {
-      return absl::InvalidArgumentError("UDS path is not null-terminated");
-    }
-    path = unix_addr->sun_path;
-  }
+  auto path = ResolvedAddrToUnixPathIfPossible(resolved_addr);
+  GRPC_RETURN_IF_ERROR(path.status());
+  std::string scheme = path->rfind('\0', 0) == 0 ? "unix-abstract" : "unix";
```

Comment:
this can be simplified to `path->at(0) == '\0'`. See https://godbolt.org/z/9G5jGWTEz. Regarding bounds-checking, I think other bits of this function would fail if path were an empty string.

Reply:
Fixed.